src/NuGet.Core/NuGet.Protocol/Plugins/PluginManager.cs

```diff
@@ -131,7 +132,7 @@ public async Task<IEnumerable<PluginDiscoveryResult>> FindAvailablePluginsAsync(
             var pluginCreationResults = new List<PluginCreationResult>();
 
             // Fast path
-            if (source.PackageSource.IsHttp && IsPluginPossiblyAvailable())
+            if (source.PackageSource.IsHttp && IsPluginPossiblyAvailable() && AreDownloadPlugingsEnabled())
```

Comment:
There could be multiple calls here, you'd ideally want the calculation of the AreDownloadPluginsEnabled to happen once.

Reply:
If we memoize it, can it be the case that throughout the life of the `PluginManager` the value of the environment variable changes and its not reflected on this method?

Reply:
We already do that with the plugin paths. 
https://github.com/NuGet/NuGet.Client/blob/e0340ccda98518f3eccf92716ffe4f283625344a/src/NuGet.Core/NuGet.Protocol/Plugins/PluginManager.cs#L332-L337